### PR TITLE
fix: add "transformsDiffer" for CATransform3D props

### PR DESCRIPTION
**Pull request by aleclarson**
## Summary

<s>Native props with `CATransform3D` type were not being processed before being diffed, which resulted in `matricesDiffer` always returning `false` after first render.</s>

Replace `matricesDiffer` comparison (for `CATransform3D` props) with new `transformsDiffer` function. The native `[RCTConvert CATransform3D:]` function has deprecated matrix arrays.

## Changelog

[iOS] [Fixed] - add "transformsDiffer" for CATransform3D props

## Test Plan

None

### CODE REVIEW

**Comment by osdnk** [comment]
Let's see

File: Libraries/Utilities/differ/transformsDiffer.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+ * @format
+ * @flow
+ */
+'use strict';
+
+const transformsDiffer = function(one, two) {
+  return one !== two ||
+    !one || !two ||
+    one.length !== two.length ||
+    JSON.stringify(one) !== JSON.stringify(two);
```
Comment (shergin) [inline Libraries/Utilities/differ/transformsDiffer.js]: Why do we need to use `JSON.stringify` here?
Which kinda values we diff here?

Comment (aleclarson) [reply Libraries/Utilities/differ/transformsDiffer.js]: An array of objects (eg: `[{ scaleX: 0.5 }, { translateX: 100 }]`) where order matters.

File: Libraries/ReactNative/getNativeComponentAttributes.js [hunk 2]
```diff
@@ -15,12 +15,13 @@ const UIManager = require('./UIManager');
 
 const insetsDiffer = require('../Utilities/differ/insetsDiffer');
 const invariant = require('invariant');
-const matricesDiffer = require('../Utilities/differ/matricesDiffer');
 const pointsDiffer = require('../Utilities/differ/pointsDiffer');
 const processColor = require('../StyleSheet/processColor');
 const processColorArray = require('../StyleSheet/processColorArray');
+const processTransform = require('../StyleSheet/processTransform');
```
Comment (philIip) [inline Libraries/ReactNative/getNativeComponentAttributes.js:21]: why do we need this import?

File: Libraries/Utilities/differ/transformsDiffer.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @format
+ * @flow
+ */
+'use strict';
+
+const transformsDiffer = function(one, two) {
```
Comment (philIip) [inline Libraries/Utilities/differ/transformsDiffer.js:12]: if both one and two are null, this would return true right? that seems not totally correct

**Review by philIip — CHANGES_REQUESTED** [review]
thanks for working on this! i have a couple of comments.

**Comment by aleclarson** [comment]
bump

**Pull request closed without merging**